File: JavaFX/05 Dades/Exemple 0500/src/main/java/com/project/AnimalsModel.java
package com.project;

public class AnimalsModel {
    private String especie;
    private int longevitat;
    private int numeropotes;

    // Constructor
    public AnimalsModel(String especie, int longevitat, int numeropotes) {
        this.especie = especie;
        this.longevitat = longevitat;
        this.numeropotes = numeropotes;
    }

    // Getters
    public String getEspecie() {
        return especie;
    }

    public int getLongevitat() {
        return longevitat;
    }

    public int getNumeropotes() {
        return numeropotes;
    }

    // Setters
    public void setEspecie(String especie) {
        this.especie = especie;
    }

    public void setLongevitat(int longevitat) {
        this.longevitat = longevitat;
    }

    public void setNumeropotes(int numeropotes) {
        this.numeropotes = numeropotes;
    }
}

File: Flutter/04 Dades/exemple0402/lib/animals_model.dart
class AnimalsModel {
  final String especie;
  final int longevitat;
  final int numeropotes;

  AnimalsModel({
    required this.especie,
    required this.longevitat,
    required this.numeropotes,
  });

  Map<String, dynamic> toMap() {
    return {
      'especie': especie,
      'longevitat': longevitat,
      'numeropotes': numeropotes,
    };
  }

  factory AnimalsModel.fromMap(Map<String, dynamic> map) {
    return AnimalsModel(
      especie: map['especie'],
      longevitat: map['longevitat'],
      numeropotes: map['numeropotes'],
    );
  }
}
